src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -255,31 +255,33 @@ public XmlObjectSerialization BuildXmlObjectSerialization(ObjectSchema objectSch
                 );
         }
 
-        private IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(PropertyBag propertyBag, SchemaObjectType objectType)
+        private IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(SerializationPropertyBag propertyBag, SchemaObjectType objectType)
         {
-            foreach (Property property in propertyBag.Properties)
+            foreach (ObjectTypeProperty property in propertyBag.Properties)
             {
-                var objectProperty = objectType.GetPropertyForSchemaProperty(property, includeParents: true);
-                var parameter = objectType.SerializationConstructor.FindParameterByInitializedProperty(objectProperty);
+                var schemaProperty = property.SchemaProperty!; // we ensured this is never null when constructing the list
```

Comment:
If this `property.SchemaPropert` is never null then why it's defined as nullable?

Reply:
we need it to be nullable because an `ObjectTypeProperty` can be either constructed from the Property from code model (in this case SchemaProperty is not null, InputModelProperty is null), or constructed from the InputModelProperty from the emitter (in this case InputModelProperty is not null, SchemaProperty is null)